modules/services/jankyborders/default.nix

```diff
@@ -0,0 +1,148 @@
+      else ["${arg}=${value}"]
+    else [];
+in {
+  meta.maintainers = [
+    maintainers.amsynist or "amsynist"
+  ];
+
+  options.services.jankyborders = {
+    enable = mkEnableOption "Enable the jankyborders service.";
+
+    package = mkPackageOptionMD pkgs "jankyborders" {};
+
+    width = mkOption {
+      type = types.float;
+      default = 5.0;
+      description = ''
+        Determines the width of the border. For example, width=5.0 creates a border 5.0 points wide.
+      '';
+    };
+
+    hidpi = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        If set to on, the border will be drawn with retina resolution.
+      '';
+    };
+
+    active_color = mkOption {
+      type = types.str;
+      default = "0xFFFFFFFF";
+      example = "0xFFFFFFFF";
+      description = ''
+        Sets the border color for the focused window (format: 0xAARRGGBB). For instance, active_color="0xff00ff00" creates a green border.
+        For Gradient Border : active_color="gradient(top_right=0x9992B3F5,bottom_left=0x9992B3F5)"
+      '';
+    };
+
+    inactive_color = mkOption {
+      type = types.str;
+      default = "0xFFFFFFFF";
+      example = "0xFFFFFFFF";
+      description = ''
+        Sets the border color for all windows not in focus (format: 0xAARRGGBB).
+        For Gradient Border : inactive_color="gradient(top_right=0x9992B3F5,bottom_left=0x9992B3F5)"
+      '';
+    };
+
+    background_color = mkOption {
+      type = types.str;
+      default = "";
+      example = "0xFFFFFFFF";
+      description = ''
+        Sets the background fill color for all windows (only 0xAARRGGBB arguments supported).
+      '';
+    };
+
+    style = mkOption {
+      type = types.str;
+      default = "round";
+      example = "square/round";
+      description = ''
+        Specifies the style of the border (either round or square).
+      '';
+    };
+
+    blur_radius = mkOption {
+      type = types.float;
+      default = 0.0;
+      example = 5.0;
+      description = ''
+        Sets the blur radius applied to the borders or backgrounds with transparency.
+      '';
+    };
+
+    ax_focus = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        If set to true, the (slower) accessibility API is used to resolve the focused window.
+      '';
+    };
+
+    blacklist = mkOption {
+      type = types.str;
+      default = "";
+      example = "Safari,kitty";
+      description = ''
+        The applications specified here are excluded from being bordered. For example, blacklist="Safari,kitty" excludes Safari and kitty from being bordered.
+      '';
+    };
+
+    whitelist = mkOption {
+      type = types.str;
+      default = "";
+      example = "Safari,kitty";
+      description = ''
+        Once this list is populated, only applications listed here are considered for receiving a border. If the whitelist is empty (default) it is inactive.
+      '';
+    };
+  };
```

Comment:
I think these should both be exposed as a list of strings (e.g. `[ "Safari" "kitty" ]`) and should be transformed into the required format inside the module. When doing so, you should also make sure to handle applications with spaces in their names (such as "Google Chrome.app").

Also, if it doesn't make sense for both a whitelist and a blacklist to be defined, then you should put an assertion in the module to check that that is not the case.

Reply:
Updated to list and handling the white spaces along with assertion to to the whitelist/blacklist options